recipes/perl/assets/get_vmware_assets.pl

```diff
@@ -0,0 +1,154 @@
+    print("--------------------------------------------------------\n");
+    print("Executing this library requires some additional libraries like \n\t'LWP' \n\t'JSON'\ \n\t'Getopt'\ \n\n");
+    print("You can specify the 'nbmaster', 'username', 'password', 'domainName', 'domainType' and 'assetsFilter' as command-line parameters\n");
+    print_usage();
+}
+
+# subroutine to process user input
+sub user_input {
+    GetOptions(
+        'nbmaster=s' => \$nbmaster,
+        'username=s' => \$username,
+        'password=s' => \$password,
+        'domainName=s' => \$domainName,
+        'domainType=s' => \$domainType,
+        'assetsFilter=s' => \$assetsFilter,
+    ) or die print_usage();
+
+    if ($nbmaster eq "") {
+        print("Please provide the value for 'nbmaster'");
+        exit;
+    }
+
+    if ($username eq "") {
+        print("Please provide the value for 'username'");
+        exit;
+    }
+
+    if ($password eq "") {
+        print("Please provide the value for 'password'");
+        exit;
+    }
+
+    $base_url = "$protocol://$nbmaster:$port/netbackup";
+}
+
+sub login {
+
+    my @argument_list = @_;
+    $base_url = $argument_list[0];
+    my $username = $argument_list[1];
+    my $password = $argument_list[2];
+    my $domainName = $argument_list[3];
+    my $domainType = $argument_list[4];
+
+    my $url = "$base_url/login";
+
+    my $req = HTTP::Request->new(POST => $url);
+    $req->header('content-type' => 'application/json');
+
+    if ($domainName eq "" && $domainType eq "") {
+        $post_data = qq({ "userName": "$username", "password": "$password" });
+    }
+    else {
+        $post_data = qq({ "domainType": "$domainType", "domainName": "$domainName", "userName": "$username", "password": "$password" });
+    }
+    $req->content($post_data);
+
+    print "\n\n**************************************************************";
+    print "\n\n Making POST Request to login to get token \n\n";
+
+    my $resp = $ua->request($req);
+    if ($resp->is_success) {
+        my $message = decode_json($resp->content);
+        $token = $message->{"token"};
+        print "Login succeeded with status code: ", $resp->code, "\n";
+    }
+    else {
+        print "Request failed with status code: ", $resp->code, "\n";
+    }
+    return $token;
+}
+
+# subroutine to get assets
+sub get_assets {
+    my $url = "$base_url/asset-service/workloads/vmware/assets";
+
+    my $req = HTTP::Request->new(GET => $url);
+    $req->header('Accept' => $content_type);
+    $req->header('Authorization' => $token);
+
+    print "\n\n**************************************************************";
+    print "\n\n Making GET Request to get assets\n\n";
+
+    my $resp = $ua->request($req);
+    if ($resp->is_success) {
+        my $message = $resp->decoded_content;
+        print "Get vmware Assets succeeded with status code: ", $resp->code, "\n\n\n";
+        printf("Below are the list of assests with associated id's \n");
```

Comment:
Why `printf` without any arguments to format? Why not plain old string interpolation?

```suggestion
        print "Below is the list of asset IDs:\n";
```

That skips the unnecessary `printf`, spells _assets_ correctly, fixes noun-verb agreement, pluralizes _ID_ more concerning, and accurately describes what's printed, which is only a list of IDs, not any assets associated with those IDs.

Reply:
to make it simple i am planning to have print "List of asset ID's: \n";